test/runnable/test42.d

```diff
@@ -5824,7 +5824,8 @@ void test4414() {
     assert(x == 7);
   }
   {
-    auto x = bytes4414()[0..4];
+    auto u = bytes4414();
+    auto x = u[0..4];
```

Comment:
Shouldn't `ubyte[4]` be infered here, instead of `ubyte[]` ?

Reply:
Not sure if you get the point of the update? The original code slices a temporary which is immediately deallocated. The new code stores the temporary so it can live alongside the slice.

Or maybe I misunderstand your point?

Reply:
Oh I get why the bug fix is needed. However, looking at this change, it seems that the compiler infers `ubyte[]` for the type of (previously) `x`, while the return of `bytes4414` is `ubyte[4]`.

My question is, should that happen ? I think not, and the fact it happens will make this deprecation more visible. In here, instead of the diff, I would have turned `auto` into `ubyte[4]`, but I don't think we should have to do that in the first place.

I get that this question is a separate concern from the bug being fixed, but as shown in this diff, it will trigger false positive in this case.

Reply:
But the return value was sliced. it should be typed as `ubyte[]`, no?

Reply:
Oh damn. Yeah you're right, and I got confused.

However, looking at the [original issue](https://issues.dlang.org/show_bug.cgi?id=4414) all the reports were slicing directly the return of the function. Walter seems to have picked up the last test case [when he fixed it](https://issues.dlang.org/show_bug.cgi?id=4414). As the fix seems to apply directly to taking an item from a static array returned by a function, this change doesn't test that anymore.

So maybe we could just test for one element ? `auto x = bytes4414()[2];` for example ?

Reply:
Yes, I see that now! I didn't see the entire context, @WalterBright?

Reply:
I don't understand what you're asking.

Reply:
```diff
-    auto x = bytes4414()[0..4];
+    auto x0 = bytes4414()[0];
+    auto x1 = bytes4414()[1];
+    auto x2 = bytes4414()[2];
+    auto x3 = bytes4414()[3];
```

And then test the values.

Reply:
> I don't understand what you're asking.

The original behavior that the `test4414` function was trying to ensure was that you could index a temporary static array, which is safe and valid. By your edits, you have fixed obviously bad behavior, but you have changed what the test is doing to something that doesn't ensure that. Arguably, it should have been written to index a single element to begin with.

Reply:
ok

Reply:
@WalterBright course of action?